object-api.go

```diff
@@ -0,0 +1,322 @@
+			return BucketInfo{}, probe.NewError(BucketNotFound{Bucket: bucket})
+		}
+		return BucketInfo{}, probe.NewError(e)
+	}
+	return BucketInfo{
+		Name:    vi.Name,
+		Created: vi.Created,
+	}, nil
+}
+
+// ListBuckets - list buckets.
+func (o objectAPI) ListBuckets() ([]BucketInfo, *probe.Error) {
+	var bucketInfos []BucketInfo
+	vols, e := o.storage.ListVols()
+	if e != nil {
+		return nil, probe.NewError(e)
+	}
+	for _, vol := range vols {
+		if !IsValidBucketName(vol.Name) {
+			continue
+		}
+		bucketInfos = append(bucketInfos, BucketInfo{vol.Name, vol.Created})
+	}
+	return bucketInfos, nil
+}
+
+// DeleteBucket - delete a bucket.
+func (o objectAPI) DeleteBucket(bucket string) *probe.Error {
+	// Verify if bucket is valid.
+	if !IsValidBucketName(bucket) {
+		return probe.NewError(BucketNameInvalid{Bucket: bucket})
+	}
+	if e := o.storage.DeleteVol(bucket); e != nil {
+		if e == errVolumeNotFound {
+			return probe.NewError(BucketNotFound{Bucket: bucket})
+		}
+		return probe.NewError(e)
+	}
+	return nil
+}
+
+/// Object Operations
+
+// GetObject - get an object.
+func (o objectAPI) GetObject(bucket, object string, startOffset int64) (io.ReadCloser, *probe.Error) {
+	// Verify if bucket is valid.
+	if !IsValidBucketName(bucket) {
+		return nil, probe.NewError(BucketNameInvalid{Bucket: bucket})
+	}
+	// Verify if object is valid.
+	if !IsValidObjectName(object) {
+		return nil, probe.NewError(ObjectNameInvalid{Bucket: bucket, Object: object})
+	}
+	r, e := o.storage.ReadFile(bucket, object, startOffset)
+	if e != nil {
+		if e == errVolumeNotFound {
+			return nil, probe.NewError(BucketNotFound{Bucket: bucket})
+		} else if e == errFileNotFound {
+			return nil, probe.NewError(ObjectNotFound{Bucket: bucket, Object: object})
+		}
+		return nil, probe.NewError(e)
+	}
+	return r, nil
+}
+
+// GetObjectInfo - get object info.
+func (o objectAPI) GetObjectInfo(bucket, object string) (ObjectInfo, *probe.Error) {
+	// Verify if bucket is valid.
+	if !IsValidBucketName(bucket) {
+		return ObjectInfo{}, probe.NewError(BucketNameInvalid{Bucket: bucket})
+	}
+	// Verify if object is valid.
+	if !IsValidObjectName(object) {
+		return ObjectInfo{}, probe.NewError(ObjectNameInvalid{Bucket: bucket, Object: object})
+	}
+	fi, e := o.storage.StatFile(bucket, object)
+	if e != nil {
+		if e == errVolumeNotFound {
+			return ObjectInfo{}, probe.NewError(BucketNotFound{Bucket: bucket})
+		} else if e == errFileNotFound || e == errIsNotRegular {
+			return ObjectInfo{}, probe.NewError(ObjectNotFound{Bucket: bucket, Object: object})
+			// Handle more lower level errors if needed.
+		} else {
+			return ObjectInfo{}, probe.NewError(e)
+		}
+	}
+	contentType := "application/octet-stream"
+	if objectExt := filepath.Ext(object); objectExt != "" {
+		content, ok := mimedb.DB[strings.ToLower(strings.TrimPrefix(objectExt, "."))]
+		if ok {
+			contentType = content.ContentType
+		}
+	}
+	return ObjectInfo{
+		Bucket:      fi.Volume,
+		Name:        fi.Name,
```

Comment:
As we consume returned `fi.Name` and `fi.Volume` as is, does it mean any `storage` should return `objectAPI` consumable values?